redis/asyncio/lock.py

```diff
@@ -163,7 +173,16 @@ async def __aenter__(self):
         raise LockError("Unable to acquire lock within the time specified")
 
     async def __aexit__(self, exc_type, exc_value, traceback):
-        await self.release()
+        try:
+            await self.release()
+        except LockNotOwnedError:
+            if self.raise_on_release_error:
+                raise
+            logger.warning("Lock was no longer owned when exiting context manager.")
+        except LockError:
+            if self.raise_on_release_error:
+                raise
+            logger.warning("Lock was unlocked when exiting context manager.")
```

Comment:
Here the suggestion I mentioned in my last comment (please see). Feel free to commit or discard it as you prefer

```suggestion
        except LockError:
            if self.raise_on_release_error:
                raise
            logger.warning("Lock was unlocked or no longer owned when exiting context manager.")
```